ts/main.ts

```diff
@@ -127,17 +182,45 @@ function renderCard(
   return $card;
 }
 
-async function renderAllCards(): Promise<void> {
+async function renderAllCardsInitial(): Promise<void> {
   await getAllSpellData();
 
+  sortCardsName();
+}
+
+renderAllCardsInitial();
+
+function sortCardsName(): void {
   for (let i = 0; i < basicSpellData.count; i++) {
     const spellInfo = basicSpellData.results[i];
     const $card = renderCard(spellInfo.name, spellInfo.level, spellInfo.url);
     $spellsListCardsDiv.appendChild($card);
   }
 }
 
-renderAllCards();
+function sortCardsLevel(): void {
+  for (let i = 0; i < spellsByLevel.length; i++) {
+    spellsByLevel[i].forEach((element) => {
+      const $card = renderCard(element.name, element.level, element.url);
+      $spellsListCardsDiv.appendChild($card);
+    });
+  }
+}
+
+$spellsListSortDropdown.addEventListener('input', () => {
+  // clear cards
+  while ($spellsListCardsDiv.childNodes.length > 0) {
+    if (!$spellsListCardsDiv.firstElementChild) break;
+    $spellsListCardsDiv.removeChild($spellsListCardsDiv.firstElementChild);
+  }
+
+  // re-render cards based on sort value
+  if ($spellsListSortDropdown.value === 'level') {
+    sortCardsLevel();
+  } else {
+    sortCardsName();
+  }
```

Comment:
We might be able to make this more efficient rather than removing every DOM element and then adding it again with two different loops. Instead if we grab all of the children and then append them again, it should move them to the new location 🤔